content/PIV/Guides/Securing_SSH_with_OpenPGP_or_PIV.adoc

```diff
@@ -0,0 +1,42 @@
+
+=== OpenPGP
+OpenPGP certificates and keys are managed on the OpenPGP function of the YubiKey. OpenPGP for SSH supports a more direct flow between the user and their key management, making it ideal for independent or open source developers who want to ensure they maintain control over their SSH keys.
+
+Instructions for setting up SSH with OpenPGP and the YubiKey are available here:
+link:../../PGP/SSH_authentication/index.adoc/[SSH Authentication]
+
+==== Advantages to OpenPGP
+OpenPGP makes it simple to manage keys on a single local machine controlled by the user. Using the OpenPGP software, users have complete control over the details of a certificate, including the key algorithm size and type, expiration date, associated user information and other details. The private key remains completely in the user’s control, existing only on the machine used to generate it and on the secure element of the YubiKey itself.
+
+Furthermore, OpenPGP makes it simple to share the corresponding public key, allowing for usage of the OpenPGP cert and key beyond just SSH. A popular usage is to have the same key for authenticating into an SSH session and for Git code signing and email encryption. Since OpenPGP has cross-platform support on Windows, Linux and MacOS, an OpenPGP key on a YubiKey can also enable SSH authentication across all platforms as well.
+
+==== Considerations when using OpenPGP ===
+As powerful as OpenPGP is, there are some drawbacks which need to be kept in mind. One of the great strengths of OpenPGP--the direct generation of a key on a local machine--can lead to loss of access if the YubiKey holding the OpenPGP key becomes unavailable. This can be mitigated if the private key is securely backed up.
+
+In larger organizations, OpenPGP’s ability to run without a greater framework also can hinder the uniform application of security policies. Without a central authority enforcing a standard set of attributes during key generation, there is a risk of a user unintentionally generating a weak key or setting an expiration date too far in the future or too soon. Again, this can be offset via training and documentation, but this mitigation strategy assumes users read the documentation.
+
+While OpenPGP is widely used for signing and encryption services, it is not commonly used for centrally managed authentication. When using certificate-based authentication, the lack of a framework to support a credential authority means OpenPGP is not supported. This and the fact that OpenPGP on a smart card is limited to a single cert and key (split into 3 sub-keys), credential management services and certificate authorities have not added native support for OpenPGP credentials.
+
+=== PIV / X.509

```

Comment:
Everything that talks about X509 certificates in the context of SSH here is confusing to me. SSH uses its own (optional) certificate format which has nothing to do with X509. Regardless of if PIV or OpenPGP is used for SSH, only the cryptographic key pair is used, any certificates are disregarded. While potentially you could connect PKI to SSH authorization via the public key, there doesn't exist a widespread solution which does this that I'm aware of. Using SSH certificates (which are not X509) can be done with either PIV or OpenPGP - it doesn't matter.